### PR TITLE
Feature/76 edit view

### PR DESCRIPTION
**Pull request by nbarlowATI**
* Some changes (e.g. frontend Dockerfile) merged in from `main`.
* Add new `lock_uuid` charfield to AssuranceCase model (replacing boolean field `editable`).
* In `CaseContainer.js` generate a new uuid and store it in both the state and the browser's sessionData (so it is maintained with "refresh").   
* Add some logic in CaseContainer to allow user to:
   -  enable edit mode if the case's `lock_uuid` is currently null,
   -  disable edit mode if the case's `lock_uuid` is currently this user's session_id
   -  override another user's edit mode if the case's `lock_uuid` is a non-null value that is not the current session_id
* Add an event listener so `lock_uuid` will be reset to null if the browser tab is closed.

### CODE REVIEW

File: frontend/src/components/CaseContainer.js [hunk 1]
```diff
@@ -347,6 +370,71 @@ class CaseContainer extends Component {
...
+      // override!
+      if (
+        window.confirm(
+          "Are you sure?  You might be overwriting someone's work..."
+        )
+      ) {
+        this.submitCaseChange("lock_uuid", this.state.session_id);
+      }
+    }
+    this.updateView();
```
Comment (mhauru) [inline frontend/src/components/CaseContainer.js:386]: I wonder if we might hit race conditions, where the browser tells the backend to set `lock_uuid` and then immediately reloads, but the backend hasn't yet had time to process the request, and sends the old value? Same question for `disableEditing`.

**Review by mhauru — APPROVED** [review]
This is great! Nice and elegant and robust seeming. I had one question about possible race conditions, can be merged as-is though.

File: frontend/src/components/CaseContainer.js (hunk 1 above)
Comment (nbarlowATI) [reply frontend/src/components/CaseContainer.js:386]: Yep, good point!   I hadn't seen this race condition in action, but it's good to be as robust as we can...  
I had a go at fixing this by putting the `updateView` inside a `then( ... )` - let me know if you think this is the right way to do it?

Comment (mhauru) [reply frontend/src/components/CaseContainer.js]: Looks good to me

**Pull request merged**